### PR TITLE
Fix #41

### LINKED ISSUE #41: Add postgres backend support for JobDB
Add a pluggable backend for postgres.

### PR DESCRIPTION
**Pull request by alpert**

### CODE REVIEW

**Comment by alpert** [comment]
Hi. I just updated request acording to your comments. Can you check please?

File: job/storage/postgres/postgres.go [hunk 1]
```diff
@@ -1,172 +1,185 @@
...
+			&meta.ErrorCount,
+			&meta.LastError,
+			&meta.LastAttemptedRun,
+			&j.NextRunAt)
+		if err != nil {
+			return nil, err
+		}
+
+		dependentJobsByte := []byte(dependentJobsStr)
+		json.Unmarshal(dependentJobsByte, &j.DependentJobs)
```
Comment (elimisteve) [inline job/storage/postgres/postgres.go]: Handle the error returned here

File: job/storage/postgres/postgres.go [hunk 2]
```diff
@@ -1,172 +1,185 @@
...
+
+		parentJobsByte := []byte(parentJobsStr)
+		json.Unmarshal(parentJobsByte, &j.ParentJobs)
```
Comment (elimisteve) [inline job/storage/postgres/postgres.go]: Handle the error returned here, too

**Comment by alpert** [comment]
Done.

**Comment by ajvb** [comment]
@alpert Sorry for the delay.

I am a big fan of this, and it looks really solid, but there are no tests and there is no storage of the `JobStat` structs associated with a `Job`. Any possibility those two things could be added?

Let me know if you need any help with either of those.

**Comment by gwoo** [comment]
Would definitely be nice to have postgres support, but still waiting for an update on `JobStat` and `more tests`.

**Pull request closed without merging**